apollo-rx2support/src/main/java/com/apollographql/android/rx2/Rx2Apollo.java

```diff
@@ -0,0 +1,148 @@
+package com.apollographql.android.rx2;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+import com.apollographql.apollo.ApolloWatcher;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.internal.util.Cancelable;
+
+import javax.annotation.Nonnull;
+
+import io.reactivex.Completable;
+import io.reactivex.CompletableEmitter;
+import io.reactivex.CompletableOnSubscribe;
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.ObservableOnSubscribe;
+import io.reactivex.Single;
+import io.reactivex.SingleEmitter;
+import io.reactivex.SingleOnSubscribe;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.exceptions.Exceptions;
+
+import static com.apollographql.apollo.api.internal.Utils.checkNotNull;
+
+/**
+ * The Rx2Apollo class provides methods for converting ApolloCall, ApolloPrefetch
+ * and ApolloWatcher types to RxJava 2 sources.
+ */
+public class Rx2Apollo {
+
+  private Rx2Apollo() {
+    throw new AssertionError("This class cannot be instantiated");
+  }
+
+  /**
+   * Converts an {@link ApolloPrefetch} to an asynchronous Observable.
```

Comment:
I think I meant ApolloWatcher here not ApolloPrefetch. But anyways even if you feel that Observable here is a sync one I present to you my argument.

So if we assume that on the client side no scheduler is being used, then when the client subscribes to this observable it will simply block waiting for emissions to take place. In that case the client would forever remain blocked because we never call onComplete in the wrapper function for ApolloWacther. The only possible way by which client would be able to move to the next statement of execution is if the other terminal event (which is onError) somehow is emitted. And I think that's not what our intention with ApolloWatcher is. We don't want clients to simply while using ApolloWatcher.